src/components/Examples/filter-search-examples.ts

```diff
@@ -0,0 +1,17 @@
+import { Example } from '../../data/examples';
+import { SelectedFilters } from './ExamplesContent';
+
+export const filterSearchExamples = (examples: Example[], selected: SelectedFilters, searchTerm: string) => {
+  const normalizedSearchTerm = searchTerm.toLowerCase();
+
+  return examples.filter(
+    (example) =>
+      (selected.products.length === 0 || example.products.some((product) => selected.products.includes(product))) &&
+      (selected.useCases.length === 0 || example.useCases.some((useCase) => selected.useCases.includes(useCase))) &&
+      (searchTerm === '' ||
+        example.name.toLowerCase().includes(normalizedSearchTerm) ||
+        example.description.toLowerCase().includes(normalizedSearchTerm) ||
+        example.products.some((product) => product.toLowerCase().includes(normalizedSearchTerm)) ||
```

Comment:
[nitpick] filtering on product has a strange effect when you search for "live" as it matches every single example and I came here to report what I thought was a bug:

https://github.com/user-attachments/assets/dda8f788-fa88-44a6-b2cd-22646813237a

Reply:
This is because all of the placeholder examples happen to be tagged with a use case of "live chat". Perhaps a UX problem, but this will be less obvious when there is a greater diversity of examples

Reply:
thought so, I just wanted to bring it to our attention before it becomes "that is how it always was" 🤣